pip/commands/list.py

```diff
@@ -110,6 +130,22 @@ def run(self, options, args):
                 "no longer has any effect.",
                 RemovedInPip10Warning,
             )
+
+        # will this work in all usage cases?
+        if '--columns' in sys.argv and '--no-columns' in sys.argv:
+            raise CommandError(
+                "Options --columns and --no-columns cannot be combined.")
+
```

Comment:
If you do the thing that I said about sharing a destination, it shouldn't matter-- argparse will take care of it.